File: backend/app/app/schemas/entry.py
import datetime
from typing import Optional

from pydantic import BaseModel

from app.schemas.tag import TagBase, TagInDBBase, TagPublic


# Shared properties
class EntryBase(BaseModel):
    content: Optional[str] = None


# Properties to receive on item creation
class EntryCreate(EntryBase):
    content: str
    date: datetime.date | None


# Properties to receive on item update
class EntryUpdate(EntryBase):
    tags: Optional[list[int]] = None


# Properties shared by models stored in DB
class EntryInDBBase(EntryBase):
    id: int
    content: str
    owner_id: int
    tags: list[TagInDBBase]
    date: datetime.date

    class Config:
        orm_mode = True


# Properties to return to client
class EntryPublic(EntryBase):
    id: int
    content: str
    tags: list[TagPublic]
    date: datetime.date | None

    class Config:
        orm_mode = True

File: frontend/src/types/entry.ts
import {TagBase} from "./tag.ts";

export type EntryBase = {
  id: number;
  content: string;
  date: Date;
  tags: TagBase[];
}

export type Entry = EntryBase & {
  meta: {
    loading: boolean;
  }
}


export type CreateEntryResponse = {
  id: number;
  content: string;
  tags: TagBase[];
  date: Date;
};

export type GetEntryResponse = CreateEntryResponse;
export type UpdateEntryResponse = CreateEntryResponse;
